pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -192,67 +163,256 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
+      const key = nativeReg[keyName];
+      const registryKey = nativeReg.openKey(key, this.registryPathProfile.join('\\'), nativeReg.Access.READ);
 
-        if (!innerKey) {
-          continue;
-        }
-        try {
-          regValue = readRegistryUsingSchema(schemaVal, innerKey);
-          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
-            // Ignore empty objects
-            regValue = null;
-          }
-        } finally {
-          nativeReg.closeKey(innerKey);
-        }
+      if (!registryKey) {
+        continue;
+      }
+      const defaultsKey = nativeReg.openKey(registryKey, DEFAULTS_HIVE_NAME, nativeReg.Access.READ);
+      const lockedKey = nativeReg.openKey(registryKey, LOCKED_HIVE_NAME, nativeReg.Access.READ);
+
+      try {
+        defaults = defaultsKey ? this.readRegistryUsingSchema(settings.defaultSettings, defaultsKey, [DEFAULTS_HIVE_NAME]) : {};
+        locked = lockedKey ? this.readRegistryUsingSchema(settings.defaultSettings, lockedKey, [LOCKED_HIVE_NAME]) : {};
+      } catch (err) {
+        console.error('Error reading deployment profile: ', err);
+      } finally {
+        nativeReg.closeKey(registryKey);
+        nativeReg.closeKey(defaultsKey);
+        nativeReg.closeKey(lockedKey);
+      }
+      // If we found something in the HKLM Defaults or Locked registry hive, don't look at the user's
+      // Alternatively, if the keys work, we could break, even if both hives are empty.
+      if (Object.keys(defaults).length || Object.keys(locked).length) {
+        break;
+      }
+    }
+
+    // Don't bother with the validator, because the registry-based reader validates as it reads.
+    return { defaults, locked };
+  }
+
+  protected fullRegistryPath(...pathParts: string[]): string {
+    return `${ this.keyName }\\${ this.registryPathProfile.join('\\') }\\${ pathParts.join('\\') }`;
+  }
+
+  /**
+   * Windows only. Read settings values from registry using schemaObj as a template.
+   * @param schemaObj the object used as a template for navigating registry.
+   * @param regKey the registry key obtained from nativeReg.openKey().
+   * @param pathParts the relative path to the current registry key, starting at 'Defaults' or 'Locked'
+   * @returns null, or the registry data as an object.
+   */
+  protected readRegistryUsingSchema(schemaObj: Record<string, any>, regKey: nativeReg.HKEY, pathParts: string[]): Record<string, any> {
+    const newObject: Record<string, any> = {};
+    const schemaKeys = Object.keys(schemaObj);
+    const commonKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    const unknownKeys: string[] = [];
+    const userDefinedObjectKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    let regValue: any;
+
+    // Drop the initial 'defaults' or 'locked' field
+    const pathPartsWithoutHiveType = pathParts.slice(1);
+
+    for (const originalKey of nativeReg.enumKeyNames(regKey)) {
```

Comment:
```suggestion
    for (const registryKey of nativeReg.enumKeyNames(regKey)) {
```
Since that makes it easier when we push to `userDefinedObjectKeys`/`commonKeys`.

Reply:
Good catch